transmogrifier/sources/transformer.py

```diff
@@ -242,30 +238,21 @@ def _transform(
             source_record: A single source record.
         """
         if self.record_is_deleted(source_record):
-            source_record_id = self.get_source_record_id(source_record)
-            timdex_record_id = self.get_timdex_record_id(
-                self.source, source_record_id, source_record
-            )
+            timdex_record_id = self.get_timdex_record_id(source_record)
             raise DeletedRecordEvent(timdex_record_id)
-        optional_fields = self.get_optional_fields(source_record)
-        if optional_fields is None:
-            return None
-
-        fields = {
-            **self.get_required_fields(source_record),
-            **optional_fields,
-        }
 
-        fields = self.create_dates_and_locations_from_publishers(fields)
-        fields = self.create_locations_from_spatial_subjects(fields)
+        timdex_record = timdex.TimdexRecord(
+            source=self.source_name,
+            source_link=self.get_source_link(source_record),
+            timdex_record_id=self.get_timdex_record_id(source_record),
+            title=self.get_valid_title(source_record),
+        )
+        for field_name, field_method in self.get_optional_field_methods():
+            setattr(timdex_record, field_name, field_method(source_record))
 
-        # If citation field was not present, generate citation from other fields
-        if fields.get("citation") is None:
-            fields["citation"] = generate_citation(fields)
-        if fields.get("content_type") is None:
-            fields["content_type"] = ["Not specified"]
+        self.generate_derived_fields(timdex_record)
```

Comment:
Hmm, I think these still need to be assigned to `timdex_record` variables? 🤔

Reply:
Re: @ghukill's comment below, this should be refactored. Even though it works, it's incongruent with the patterns we've been using so far